cedar-policy-core/src/est/expr.rs

```diff
@@ -1048,15 +1075,11 @@ impl TryFrom<&Node<Option<cst::Unary>>> for Expr {
                         // not safe to collapse `--` to nothing
                         Ok(Expr::neg(Expr::neg(inner)))
                     }
-                    n => {
-                        if n % 2 == 0 {
-                            // safe to collapse to `--` but not to nothing
-                            Ok(Expr::neg(Expr::neg(inner)))
-                        } else {
-                            // safe to collapse to -
-                            Ok(Expr::neg(inner))
-                        }
-                    }
+                    3 => Ok(Expr::neg(Expr::neg(Expr::neg(inner)))),
+                    4 => Ok(Expr::neg(Expr::neg(Expr::neg(Expr::neg(inner))))),
+                    _ => Err(u
+                        .to_ast_err(ToASTErrorKind::UnaryOpLimit(ast::UnaryOp::Neg))
+                        .into()),
```

Comment:
Looks like we're not enforcing the unary op limit in the bang case above, should we be enforcing it here?

Reply:
It's implicit in that it's assumed that `num_bangs` is going to be less than 4. In my opinion I'd like to change to it from a `usize` to an enum, so that we can ensure the domain is correct.

Reply:
But that's probably too big of a change for this PR